onnx/large_helper.py

```diff
@@ -0,0 +1,333 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import enum
+import os
+from typing import Any, Dict, Iterable, Optional, Set, Tuple
+
+import numpy as np
+
+from onnx import (
+    GraphProto,
+    ModelProto,
+    StringStringEntryProto,
+    TensorProto,
+    checker,
+    load_model,
+)
+from onnx.external_data_helper import (
+    ExternalDataInfo,
+    _get_all_tensors,
+    _sanitize_path,
+    uses_external_data,
+)
+from onnx.helper import make_model
+
+
+def make_large_tensor_proto(
+    location: str, name: str, tensor_type: int, shape: Tuple[int, ...]
+) -> TensorProto:
+    """
+    Create an external tensor.
+
+    :param location: unique identifier (not necessary a path)
+    :param name: initializer name in the graph
+    :param tensor_type: onnx type
+    :param shape: shape the of the initializer
+    :return: the created tensor
+    """
+    tensor_location = location
+    tensor = TensorProto()
+    tensor.name = name
+
+    del tensor.external_data[:]
+    tensor.data_location = TensorProto.EXTERNAL
+    for k, v in {
+        "location": tensor_location,
+        "offset": None,
+        "length": None,
+        "checksum": None,
+        "basepath": None,
+    }.items():
+        if v is not None:
+            entry = tensor.external_data.add()
+            entry.key = k
+            entry.value = str(v)
```

Comment:
Can this be simplified and dictionary avoided?

Reply:
It is copy/pasted from external_data_helper but I needed the same function without the verification on the location. I was hesitating, should I reuse the same function but add an argument to disable the exception? (set_external_data)

Reply:
nit: this can be simplified, since the None fields have no effect. So, it is really only one field, "location", so no need for a loop. But I would go further and not even set the location field. Why can't we just use the tensor-name as the key, why do we need this location field?

Reply:
I am talking about just the in-memory representation. When serialized/saved, these fields will be set as per ONNX spec.

Reply:
I simplified.